lib/utils/DirectUpload/index.js

```diff
@@ -0,0 +1,51 @@
+import directUploadsApi from "apis/directUploads";
+import { noop } from "utils/common";
+
+import { generateChecksum } from "./utils";
+
+class DirectUpload {
+  constructor({ url, file, progress = noop }) {
+    this.url = url;
+    this.file = file;
+    this.progress = progress;
+    this.abortController = new AbortController();
+  }
+
+  create = async () => {
+    try {
+      const { data } = await this.generateUrl();
+      const {
+        direct_upload: { url, headers },
+      } = data;
+      await this.uploadToCloud(url, headers);
+      return Promise.resolve(data);
```

Comment:
Same comment as above. should be possible to return data as it is. please verify and check for other similar occurrences